src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -2521,19 +2521,28 @@ pbs_python_populate_svrattrl_from_python_class(PyObject *py_instance,
 	PyObject	*py_resc_hookset_dict0 = (PyObject *)NULL;
 	PyObject	*py_attr_keys = (PyObject *)NULL;
 	PyObject 	*py_val = (PyObject *)NULL;
-	char	*name_str_dup = NULL;
-	char	*val_str_dup = NULL;
+	char		*name_str_dup = NULL;
+	char		*val_str_dup = NULL;
 	int		num_attrs, i;
 	pbs_list_head	svrattrl_list2;
-	int         rc = -1;
+	int		rc = -1;
 	int		hook_set_flag = 0;
 	int		has_resv_duration;
-	char            the_resc[HOOK_BUF_SIZE];
-	char            the_val[HOOK_BUF_SIZE];
+	char		the_resc[HOOK_BUF_SIZE];
+	static char     *the_val = NULL;
+	static int 	val_buf_size = HOOK_BUF_SIZE;
 	PyObject	*py_resc = (PyObject *)NULL;
 	long		val_sec;
 	char		*objname = NULL;
 
+	if (the_val == NULL) {
+		the_val = (char *)malloc(HOOK_BUF_SIZE*sizeof(char));
+		if (the_val == NULL) {
+			log_err(PBSE_SYSTEM, __func__, "Out of memory error");
+			return rc;
+		}
+	}
+
```

Comment:
Kindly look at the implementation of pbs_strcat(), we can pass address of a pointer that is assigned NULL to it. We need not pass address of an already allocated buffer. 
So, we can remove the statement on line 2738.
The realloc() inside pbs_strcat() will take care of the memory allocation and the logic on line 2738 is also taken care of in pbs_strcat().
Hence, we can initialize the pointer the_val to NULL and val_buf_size to zero, remove the malloc() and rest will be taken care of in pbs_strcat().